src/Application.vala

```diff
@@ -133,20 +133,13 @@ public class ValaLint.Application : GLib.Application {
         }
 
         /* 4. Print mistakes */
-        print_mistakes (file_data_list);
+        bool has_errors = print_mistakes (file_data_list);
 
         if (exit_with_zero) {
             return 0;
         }
 
-        foreach (FileData file_data in file_data_list) {
-            foreach (FormatMistake? mistake in file_data.mistakes) {
-                if (mistake.check.state == Config.State.ERROR) {
-                    return 1;
-                }
-            }
-        }
-        return 0;
+        return (int) has_errors;
```

Comment:
I do not think this is good Vala style.  You could combine with the `exit_with_zero` test:
```
if (exit_with_zero || !has_errors) {
    return 0;
} else {
    return 1;
}

Reply:
Updated